synapse/storage/databases/main/devices.py

```diff
@@ -1045,6 +1050,63 @@ def _txn(txn):
 
         return rows
 
+    async def _remove_deleted_devices_from_device_inbox(
+        self, progress: dict, batch_size: int
+    ) -> int:
+        """A background update that deletes all device_inboxes for deleted devices.
+
+        This should only need to be run once (when users upgrade to v1.45.0)
```

Comment:
There are some delta sql scripts mentioned below. Is the intent we run those as migrations on synapse startup to trigger this bg update? (Rather than admins manually triggering it)

Reply:
Yes, this is for migration. The intent ist to run in bg job instead waiting for startup because it need a lot of time to run:
```sql
DELETE FROM device_inbox WHERE device_id NOT IN (SELECT device_id FROM devices);
```

IMO it is an improvment to #10098
Similar to this discussion: https://github.com/matrix-org/synapse/pull/10097#discussion_r643171952 and this PR: #10581